src/Trakx.CoinGecko.ApiClient/ClientRegistrations.tt

```diff
@@ -0,0 +1,51 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ output extension=".cs" #>
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Polly;
+using Polly.Contrib.WaitAndRetry;
+using Polly.Extensions.Http;
+using Serilog;
+
+<# 
+var nameSpace = "Trakx.CoinGecko.ApiClient";
+#>
+namespace <#= nameSpace #>
+{
+    public static partial class AddCoinGeckoClientExtension
+    {
+        private static void AddClients(this IServiceCollection services)
+        {
+            var delay = Backoff.DecorrelatedJitterBackoffV2(medianFirstRetryDelay: TimeSpan.FromMilliseconds(100), retryCount: 10, fastFirst: true);
+            <# 
+            var clientNames = new [] { /*"ExchangesClient",*/ "MarketDataClient", "AccountsClient" };
```

Comment:
Here you have a few more clients
![image](https://user-images.githubusercontent.com/4638821/113825432-dc21bb80-9778-11eb-8566-5c8b728cbc85.png)
I think you can list them, and also get a few bonus things to get records instead of classes by looking at this adding a new class like this one

```
using System.IO;
using Trakx.Utils.Extensions;
using Xunit.Abstractions;

namespace Trakx.Circle.ApiClient.Tests.Integration
{
    public class OpenApiGeneratedCodeModifier : Trakx.Utils.Testing.OpenApiGeneratedCodeModifier
    {
        public OpenApiGeneratedCodeModifier(ITestOutputHelper output)
            : base(output)
        {
            var foundRoot = default(DirectoryInfo).TryWalkBackToRepositoryRoot(out var rootDirectory)!; 
            FilePaths.Add(Path.Combine(rootDirectory!.FullName, "src",
                "Trakx.CoinGecko.ApiClient", "ApiClients.cs"));
        }
    }
}

```

Really sorry, maybe I should have told you to use the Circle client as base because it seems to be more up to date...

Reply:
No worries. I will compare Shrimpy and Circle codes and apply the updates to my new code.

Reply:
Done